packages/amplify-graphql-auth-transformer/src/resolvers/subscriptions.ts

```diff
@@ -30,38 +37,100 @@ import {
   setHasAuthExpression,
   generateOwnerClaimExpression,
   generateOwnerClaimListExpression,
+  getIdentityClaimExp,
+  getOwnerClaimReference,
 } from './helpers';
 
 const dynamicRoleExpression = (roles: Array<RoleDefinition>): Array<Expression> => {
+  const dynamicExpression = new Array<Expression>();
   const ownerExpression = new Array<Expression>();
+  const groupExpression = new Array<Expression>();
   // we only check against owner rules which are not list fields
   roles.forEach((role, idx) => {
     if (role.strategy === 'owner') {
+      const ownerClaimRef = getOwnerClaimReference(role.claim!, `ownerClaim${idx}`);
       ownerExpression.push(
         generateOwnerClaimExpression(role.claim!, `ownerClaim${idx}`),
         generateOwnerClaimListExpression(role.claim!, `ownerClaimsList${idx}`),
+        qref(methodCall(ref('authOwnerRuntimeFilter.add'), raw(`{ "${role.entity}": { "${role.isEntityList ? 'contains' : 'eq'}": $${ownerClaimRef} } }`))),
         set(
           ref(`ownerEntity${idx}`),
           methodCall(ref('util.defaultIfNull'), ref(`ctx.args.${role.entity!}`), nul()),
         ),
         iff(
-          not(ref(IS_AUTHORIZED_FLAG)),
+          and([
+            not(ref(IS_AUTHORIZED_FLAG)),
+            not(methodCall(ref('util.isNullOrEmpty'), ref(`ownerEntity${idx}`))),
+          ]),
           compoundExpression([
-            iff(
+            ifElse(
               or([
                 equals(ref(`ownerEntity${idx}`), ref(`ownerClaim${idx}`)),
                 methodCall(ref(`ownerClaimsList${idx}.contains`), ref(`ownerEntity${idx}`)),
               ]),
-              set(ref(IS_AUTHORIZED_FLAG), bool(true))),
+              set(ref(IS_AUTHORIZED_FLAG), bool(true)),
+              methodCall(ref('util.unauthorized')),
```

Comment:
If customer has passed an invalid owner argument, we will reject the client connection immediately.